### PR TITLE
refactor: only get credentials when required

### LINKED ISSUE #173: Account Login Suppression
## Expected Behaviour

Executor actions that do not require accessing the cloud are not held up by a need to login to the cloud provider account CLI.

## Current Behaviour

In the process of running `setContext.sh` which occurs early in most scripts in the Executor, the cloud provider runs setCredentials.sh - this evaluates the current account and provider and runs the login applicable login commands.

## Possible Solution

- add an option on the setContext.sh script that when provided, skips the setCredentials.sh call.
- review existing uses of setContext.sh and add this option to its args for any call that is not interacting with the cloud account.

## Context

perform basic local Hamlet Deploy tasks with no intention of interacting with the cloud provider or account.

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description

In the bash executor cli only get the credentials for cloud providers when they are required

With the use of more query tooling in the hamlet cli we generate templates which are used by a hamlet user to find out about thier deployment

## Motivation and Context

Fixes #173 

## How Has This Been Tested?

Tested locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

File: cli/createTemplate.sh [hunk 1]
```diff
@@ -978,7 +981,7 @@ function process_template() {
     process_template_pass \
       "${entrance}" \
       "${flows}" \
-      "${task_parameters[@]}" \
+      "${task_parameters[@]:0:7}" \
```
Comment (ml019) [inline cli/createTemplate.sh:984]: IS this related to credentials? Looks some insurance against too many parameters being provided?

Comment (roleyfoley) [reply cli/createTemplate.sh:984]: Sorry it snuck in and forgot to mention it. Yes I've been working on extending the generation contract to include more details which means that for now we only want particular values

**Review by ml019 — APPROVED** [review]
